drivers/sensor/bmm150/bmm150.c

```diff
@@ -0,0 +1,595 @@
+/* bmm150.c - Driver for Bosch BMM150 Geomagnetic Sensor */
+
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * This code is based on bmm050.c from:
+ * https://github.com/BoschSensortec/BMM050_driver
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "bmm150.h"
+
+static const struct {
+	int freq;
+	u8_t reg_val;
+} bmm150_samp_freq_table[] = { {2, 0x01},
+			       {6, 0x02},
+			       {8, 0x03},
+			       {10, 0x00},
+			       {15, 0x04},
+			       {20, 0x05},
+			       {25, 0x06},
+			       {30, 0x07} 
+};
+
+static const struct bmm150_preset {
+	u8_t rep_xy;
+	u8_t rep_z;
+	u8_t odr;
+} bmm150_presets_table[] = {
+	[LOW_POWER_PRESET] = {3 , 3, 10},
+	[REGULAR_PRESET] = {9, 15, 10},
+	[ENHANCED_REGULAR_PRESET] = {15, 27, 10},
+	[HIGH_ACCURACY_PRESET] = {47, 83, 20}
+};
+
+static int bmm150_set_power_mode(struct device *dev,
+				enum bmm150_power_modes mode,
+				int state)
+{
+	struct bmm150_data *data = dev->driver_data;
+	const struct bmm150_config *config = dev->config->config_info;
+
+	switch(mode){
+		case BMM150_POWER_MODE_SUSPEND:
+			if(i2c_reg_update_byte(data->i2c,
+						config->i2c_slave_addr,
+						BMM150_REG_POWER,
+						BMM150_MASK_POWER_CTL,
+						!state) < 0) {
+				return -EIO;
+			}
+			k_busy_wait(5 * USEC_PER_MSEC);
```

Comment:
OK. Removed this delay as of no. I took the reference from another zephyr driver but not needed now